src-electron/db/query-config.js

```diff
@@ -211,15 +212,18 @@ async function insertOrUpdateAttributeState(
       attributeId,
       clusterRef
     )
-  if (
-    staticAttribute.storagePolicy ==
-    dbEnum.storagePolicy.attributeAccessInterface
-  ) {
-    staticAttribute.storagePolicy = dbEnum.storageOption.external
-  } else {
-    staticAttribute.storagePolicy = dbEnum.storageOption.ram
-  }
-
+  let forcedExternal = await queryUpgrade.getForcedExternalStorage(
+    db,
+    attributeId
+  )
+  let storageOption = await queryUpgrade.computeStorage(
+    db,
+    null,
+    clusterRef,
+    staticAttribute.storagePolicy,
+    forcedExternal,
```

Comment:
I guess my architectural question is this: if we are doing the "getForcedExternalStorage" thing here, what is the point of having storagePolicy as a concept at all?

That is, why are we bothering to store a storage policy in the db?

All of this could really benefit from some architecture documentation.

Reply:
Going into this I had no idea what the history behind storagePolicy was. After debugging, I think we have to do getForcedExternalStorage here because it is the only way to reference the cluster. 

Note, we are still using storage policy when creating endpoints from scratch. As in, when we have a "new Configuration". But point taken.